src/common/snippets/include/snippets/op/tile_scheduler.hpp

```diff
@@ -0,0 +1,40 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "ngraph/op/op.hpp"
+#include "snippets/emitter.hpp"
+#include "tile.hpp"
+
+namespace ngraph {
+namespace snippets {
+namespace op {
+
+/**
+ * @interface TileScheduler
+ * @brief Contains a set of Tiles (currently one vector and one scalar) and performs necessary preparations
+ * before the Tiles could be executed: calculates offsets, sets proper work amounts, decrement pointers if the same data
+ * have to be read several times (broadcasting).
+ * @ingroup snippets
+ */
+class TileScheduler : public ngraph::op::Op {
+public:
+    OPENVINO_OP("TileScheduler", "SnippetsOpset");
+
+    TileScheduler(const std::pair<std::shared_ptr<ngraph::snippets::Emitter>, ngraph::snippets::RegInfo> &vector_region,
+                  const std::pair<std::shared_ptr<ngraph::snippets::Emitter>, ngraph::snippets::RegInfo> &scalar_region);
+    TileScheduler() = default;
+    std::pair<std::shared_ptr<ngraph::snippets::Emitter>, ngraph::snippets::RegInfo> vector_region;
+    std::pair<std::shared_ptr<ngraph::snippets::Emitter>, ngraph::snippets::RegInfo> scalar_region;
```

Comment:
I think so. Vector & scalar regions are public because we need to access them from TileSchedulerEmitter constructor. Of course we could make them private and provide a getter methods, but I believe this is an unnecessary complication at this point.